### PR TITLE
Use torch.multiprocessing only if installed

### PR DESCRIPTION
**Pull request by ilia-tsyplenkov**
## Description

Please include a summary of the change and the issue it solves. 

**Fixes** #(issue)

## Type of change

(Please delete options that are not relevant)

- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [ ] ⚡ New feature (non-breaking change which adds functionality)
- [ ] 📢 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📄 This change requires a documentation update

### What is the solution?

(Describe at a high level how the feature was implemented)

## Checklist:

- [ ] My code follows the style guidelines(PEP 8) of MindsDB.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have updated the documentation, or created issues to update them.
- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] I have shared a short loom video or screenshots demonstrating any new functionality.

### CODE REVIEW

File: mindsdb/__main__.py [hunk 1]
```diff
@@ -29,9 +29,13 @@
...
 from mindsdb.utilities.auth import register_oauth_client, get_aws_meta_data
 
-import torch.multiprocessing as mp
 try:
+    import torch.multiprocessing as mp
     mp.set_start_method('spawn')
+    os.environ["MINDSDB_USE_TORCH"] = "1"
+except (ImportError, ModuleNotFoundError):
+    log.logger.info("torch is not installed. torch.multiprocessing won't be used")
+    import multiprocessing as mp
```
Comment (AndrewFarley) [inline mindsdb/__main__.py:38]: NOTE: There's a bunch of commands below in "init" that use mp.* commands, which means they will fail I believe.  We need them to find alternatives.

**Review by AndrewFarley — CHANGES_REQUESTED** [review]
Needs work, see comments

File: mindsdb/api/http/start.py [hunk 2]
```diff
@@ -1,6 +1,9 @@
 import logging
 import os
-import torch.multiprocessing as mp
+try:
+    import torch.multiprocessing as mp
+except (ImportError, ModuleNotFoundError):
+    import multiprocessing as mp
```
[the remark this answers is not in the document]
Comment (ilia-tsyplenkov) [reply mindsdb/api/http/start.py:6]: torch.multiprocessing has same public API as python.multiprocessing. https://docs.python.org/3/library/multiprocessing.html#multiprocessing.cpu_count

File: mindsdb/integrations/libs/ml_exec_base.py [hunk 3]
```diff
@@ -54,7 +54,11 @@
 
 from .ml_handler_proc import MLHandlerWrapper, MLHandlerPersistWrapper
 
-import torch.multiprocessing as mp
+try:
+    import torch.multiprocessing as mp
+except (ImportError, ModuleNotFoundError):
+    import multiprocessing as mp
```
[the remark this answers is not in the document]
Comment (ilia-tsyplenkov) [reply mindsdb/integrations/libs/ml_exec_base.py:60]: torch.multiprocessing has same public API as python.multiprocessing. https://docs.python.org/3/library/multiprocessing.html#multiprocessing.get_context

File: mindsdb/integrations/utilities/processes.py [hunk 4]
```diff
@@ -1,4 +1,7 @@
-import torch.multiprocessing as mp
+try:
+    import torch.multiprocessing as mp
+except (ImportError, ModuleNotFoundError):
+    import multiprocessing as mp
```
[the remark this answers is not in the document]
Comment (ilia-tsyplenkov) [reply mindsdb/integrations/utilities/processes.py:4]: torch.multiprocessing has same public API as python.multiprocessing. https://docs.python.org/3/library/multiprocessing.html#multiprocessing.get_context

**Comment by ZoranPandovski** [comment]
@tomhuds Should we close this?

**Comment by ZoranPandovski** [comment]
Closing as outdated

**Pull request closed without merging**